bundles/persistence/org.openhab.persistence.jdbc/java/org/openhab/persistence/jdbc/internal/JdbcConfiguration.java

```diff
@@ -68,39 +70,155 @@ public JdbcConfiguration(Map<Object, Object> configuration) {
     /**
      * @{inheritDoc
      */
-    public void updateConfig(Map<Object, Object> configuration) {
+    public void updateConfig(Map<Object, Object> config) {
+        configuration = config;
+
         logger.debug("JDBC::updateConfig: configuration.size = " + configuration.size());
 
-        // Database-Url jdbc:h2:./testH2
+        // mandatory user
+        String user = (String) configuration.get("user");
+
+        // mandatory password
+        String password = (String) configuration.get("password");
+
+        // mandatory url
         String url = (String) configuration.get("url");
         Properties parsedURL = StringUtilsExt.parseJdbcURL(url);
-        logger.debug("JDBC::updateConfig: url={}", url);
-        if (StringUtils.isBlank(url) || parsedURL.getProperty("parseValid") == "false") {
-            logger.error(
-                    "JDBC::updateConfig: url The SQL database URL is missing - please configure the jdbc:url parameter in openhab.cfg");
+
+        if (user == null || StringUtils.isBlank(user)) {
+            logger.warn(
+                    "JDBC::updateConfig: SQL user is missing - please configure the jdbc:user parameter in openhab.cfg");
+        } else if (password == null || StringUtils.isBlank(password)) {
+            logger.warn("JDBC::updateConfig: SQL password is missing. Attempting to connect without password. "
+                    + "To specify a password configure the jdbc:password parameter in openhab.cfg.");
+        } else if (!(url == null || StringUtils.isBlank(url) || parsedURL.getProperty("parseValid") == "false")) {
```

Comment:
Don't use == for string equality tests, as it may only accidentally work.  Is `.equalsIgnoreCase("false")` more suitable here?